src/main/java/org/spongepowered/api/event/entity/ChangeEntityExperienceEvent.java

```diff
@@ -24,33 +24,70 @@
  */
 package org.spongepowered.api.event.entity;
 
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableExperienceHolderData;
+import org.spongepowered.api.data.manipulator.mutable.entity.ExperienceHolderData;
+import org.spongepowered.api.entity.living.player.Player;
 import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.util.annotation.eventgen.AbsoluteSortPosition;
+import org.spongepowered.api.util.annotation.eventgen.PropertySettings;
 
 /**
  * An event that is related to experience.
  */
 public interface ChangeEntityExperienceEvent extends TargetEntityEvent, Cancellable {
 
+    @Override
+    Player getTargetEntity();
```

Comment:
Target entity or target player?

Reply:
Yeah if we're doing this why doesn't it just extend `TargetPlayerEvent` instead?